### PR TITLE
Feedstock rc branch management in do-nothing script

### PR DESCRIPTION
**Pull request by trexfeathers**
### Description

---
[Consult Iris pull request check list]( https://scitools-iris.readthedocs.io/en/latest/developers_guide/contributing_pull_request_checklist.html)

### CODE REVIEW

File: tools/release_do_nothing.py [hunk 1]
```diff
@@ -526,7 +526,33 @@ def update_conda_forge(
     _wait_for_done(message)
 
     if is_release_candidate:
-        upstream_branch = "release-candidate"
+        # We chose this odd handling of release candidate branches because
```
Comment (pp-mo) [inline tools/release_do_nothing.py:529]: I'm probably missing something here, but it seems now to say this new part about "renaming the old branch" **_every_** time we create a release candidate.
Can it not be made to happen only when there _is_ a pre-existing one ?

**Review by pp-mo — CHANGES_REQUESTED** [review]
Generally looks reasonable, no other problems I can seee

**Review by bjlittle — APPROVED** [review]
@trexfeathers LGTM 👍

File: tools/release_do_nothing.py (hunk 1 above)
Comment (trexfeathers) [reply tools/release_do_nothing.py]: I agree this needed more nuance, so I've had another go

cbee22fd09a15e7a6b11dbc3ea7f6769b6b98497

**Review by pp-mo — APPROVED** [review]

**Comment by trexfeathers** [comment]
Many thanks @bjlittle and @pp-mo 🥂

**Pull request merged**